src/teachy_tv.c

```diff
@@ -0,0 +1,1775 @@
+{
+    s16 *data;
+    struct Sprite *v3;
+
+    data = gTasks[taskId].data;
+    v3 = &gSprites[data[1]];
+    StartSpriteAnim(v3, 6);
+    ++(u16)data[3];
+    data[4] = 0;
+    data[5] = 0;
+    TeachyTvGrassAnimationMain(taskId, v3->pos2.x, v3->pos2.y, 0, 0);
+}
+
+void TeachyTvClusFuncDudeMoveLeft(u8 taskId)
+{
+    s16 *data;
+    struct Sprite *v2;
+
+    data = gTasks[taskId].data;
+    v2 = &gSprites[data[1]];
+
+    if ( !(v2->pos2.x & 0xF) )
+        TeachyTvGrassAnimationMain(taskId, v2->pos2.x - 8, v2->pos2.y, 0, 0);
+    if ( v2->pos2.x == 8 )
+        ++(u16)data[3];
+    else
+        --v2->pos2.x;
+}
+
+void TeachyTvClusFuncRenderAndRemoveBg1EndGraphic(u8 taskId)
+{
+    s16 *data;
+    int temp;
+
+    data = gTasks[taskId].data;
+    if ( !data[2] )
+    {
+        CopyToBgTilemapBufferRect_ChangePalette(1, &gUnknown_8479590, 0x14, 0xA, 8, 2, 0x11);
+        schedule_bg_copy_tilemap_to_vram(1);
+    }
+    temp = (u16)data[2] + 1;
+    data[2] = temp;
+    if ( (s16)temp > 126 )
+    {
+        TeachyTvClearBg1EndGraphicText();
+        (u16)data[2] = 0;
+        ++(u16)data[3];
+    }
+}
+
+void TeachyTvClearBg1EndGraphicText()
+{
+    FillBgTilemapBufferRect_Palette0(1, 0, 0x14, 0xA, 8, 2);
+    schedule_bg_copy_tilemap_to_vram(1);
+}
+
+void TeachyTvBackToOptionList(u8 taskId)
+{
+    s16 *data;
+    s32 temp;
+
+    data = gTasks[taskId].data;
+    if ( !data[2] )
+        PlayNewMapMusic(BGM_FRLG_TEACHY_TV);
+    TeachyTvBg2AnimController();
+    temp = (u16)data[2] + 1;
+    data[2] = temp;
+    if ( (s16)temp > 0x3F )
+    {
+        data[2] = 0;
+        data[3] = 0;
+        *data = TeachyTvSetupWindow();
+        gTasks[taskId].func = TeachyTvOptionListController;
+        PutWindowTilemap(0);
+        TeachyTvSetupScrollIndicatorArrowPair();
+        TeachyTvSetWindowRegs();
+        schedule_bg_copy_tilemap_to_vram(0);
+        ChangeBgX(3, 0x0, 0);
+        ChangeBgY(3, 0x0, 0);
+        ChangeBgX(3, 0x1000, 2);
+        ChangeBgY(3, 0x2800, 1);
+        gUnknown_203F450->var_4004 = 0;
+        gUnknown_203F450->var_4005 = 3;
+        gUnknown_203F450->var_4006 = 0;
+    }
+}
+
+void TeachyTvChainTaskBattleOrFadeByOptionChosen(u8 taskId)
+{
+    int op = gTeachyTV_StaticResources.optionChosen;
+    if ( op < 0 )
+        return;
+    if ( op <= 3 )
+    {
+        TeachyTvPrepBattle(taskId);
+
+    }
+    else if ( op <= 5 )
+    {
+        *((void(**)())gUnknown_203F450) = TeachyTvSetupBagItemsByOptionChosen;
+        TeachyTvQuitBeginFade(taskId);
+    }
+}
+
+void TeachyTvSetupBagItemsByOptionChosen()
+{
+    if ( gTeachyTV_StaticResources.optionChosen == 4 )
+        sub_810B108(10);
+    else
+        sub_810B108(9);
+}
+
+void TeachyTvPostBattleFadeControl(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    if ( !(gPaletteFade.active) )
+    {
+        const u8* funcIdx = &gUnknown_8479390;
+        int arg = funcIdx[gTeachyTV_StaticResources.optionChosen];
+        data[3] = arg;
```

Comment:
`data[3] = gUnknown_8479390[gTeachyTV_StaticResources.optionChosen];`

Reply:
I guess you mean `data[3] = (&gUnknown_8479390)[gTeachyTV_StaticResources.optionChosen];`, but it didn't work (similar to the previous `text` pointer issue)

Reply:
`gUnknown_8479390` should be an array by itself